cherrypy/_cpwsgi_server.py

```diff
@@ -89,23 +89,26 @@ def __init__(self, server_adapter=cherrypy.server):
             ssl_module = self.server_adapter.ssl_module or 'builtin'
         else:
             ssl_module = self.server_adapter.ssl_module or 'pyopenssl'
+
+        ssl_args = [
+            self.server_adapter.ssl_certificate,
+            self.server_adapter.ssl_private_key,
+            self.server_adapter.ssl_certificate_chain,
+            self.server_adapter.ssl_ciphers,
+        ]
+        ssl_kwargs = {}
+        if self.server_adapter.ssl_private_key_password is not None:
+            ssl_kwargs['private_key_password'] = (
+                self.server_adapter.ssl_private_key_password
+            )
+
         if self.server_adapter.ssl_context:
             adapter_class = cheroot.server.get_ssl_adapter_class(ssl_module)
```

Comment:
Perhaps, we could collapse these branches and then get rid of the additional args?
```suggestion
        set_up_tls = self.server_adapter.ssl_context or self.server_adapter.ssl_certificate
        if set_up_tls:
            adapter_class = cheroot.server.get_ssl_adapter_class(ssl_module)
            self.ssl_adapter = adapter_class(*ssl_args, **ssl_kwargs)  # FIXME: drop the indirectly unwrapped args in favor of explicit
            if self.server_adapter.ssl_context:
                self.ssl_adapter.context = self.server_adapter.ssl_context
```

Reply:
Done as suggested. There was duplicate code, so with this change it is much better.

Reply:
Marking as resolved. Will leave comments on the new diff.